packages/git/src/browser/git-quick-open-service.ts

```diff
@@ -77,11 +77,16 @@ export class GitQuickOpenService {
                     dynamicItems.push(new SingleStringInputOpenItem(
                         `Clone the Git repository: ${lookFor}. ${suffix}`,
                         async () => {
+                            const progress = await gitQuickOpenService.messageService.showProgress({
```

Comment:
cf. links in:
https://github.com/eclipse/che/issues/12564#issuecomment-459344603
https://github.com/eclipse/che/issues/12564#issuecomment-459344937

it was discussed couple of times before. and it would really make sense so tackle this in a more generic way.

Reply:
Ok. There's more general issue https://github.com/theia-ide/theia/issues/759.

What about approve improvements for inProgress notifications only?
We are using them in our API and it will be nice to improve the existing behavior.  In this case, I will remove the first commit and change the PR description